Remotion/Web/Development.WebTesting/ScreenshotCreation/BrowserContentLocators/EdgeBrowserContentLocator.cs

```diff
@@ -153,5 +155,21 @@ private TResult RetryUntilValueChanges<TResult> (Func<TResult> func, TResult val
 
       return value;
     }
+
+    private AutomationElement GetElementWithLargerArea (AutomationElement firstAutomationElement, AutomationElement secondAutomationElement)
+    {
+      var firstRectangle = firstAutomationElement.Current.BoundingRectangle;
+      var secondRectangle = secondAutomationElement.Current.BoundingRectangle;
+
+      if (firstRectangle == Rect.Empty)

```

Comment:
aren't those two redundant and actually introduce more complexity given they are covered by the condition in line 170?

also, should be make two variables for the areas (firstRectangleArea, secondRectangleArea)? Am not sure, it just may be easier to read.

Why do we prefer the second element when they're equal? This seems to be a new condition, so there should be a logical reason for this (i.e. the last element is also the leafiest elemtn of the tree and thus the most relevant). Otherwise, I think I might prefer the first one, strictly based on the concept of stable ordering algorithms.

Reply:
> aren't those two redundant and actually introduce more complexity given they are covered by the condition in line 170?

This statement would be correct for `System.Drawing.Rectangle`, but in the wonderful world of `System.Windows.Rect` the `Empty's` height and width are not `0` but `-Infinity`, making their area `Infinity` which is naturally larger than any other area, so we have to handle these special cases separately.

Since you had to ask, it might make sense to add a code comment for this peculiar situation.

And no, we cannot rely on their leafiness, I'll just return the first one if they're equal.